### PR TITLE
Complete overhaul of major tilesets + extras

### PR DESCRIPTION
**Pull request by Rusty-Box**
This PR provides heavy changes to all major level assets from both **Icy Island** and **Rooted Forest**, as well as re-introduce the old ghost forest theme as the new **Corrupted Forest**.

---

**Initial changes include the following:**
- Revamped forest tileset
- Revamped underground forest tileset
- New corrupted forest tileset
- New corrupted underground forest tileset
- New Ghost Tree roots tileset
- Updated trees for both normal and corrupted forest
- Updated background assets for both normal and corrupted forest
- Updated spikevine graphics
- Autotiling for all major forest tilesets
- New goal houses for both normal and corrupted forest
- Remove all the unneeded/outdated graphics

**With patch 1, the following was also added:**
- Updated snow surface tileset
- Improved ice cave tileset
- Theme specific block overlays
- Updated castle tilesets
- New Ruins tileset
- Wall torch variant for torch object
- Floor tile for goal post
- Normal and corrupted flowers for the forest theme
- New sign to indicate the requirement of bombs
- Removal of more old/unneeded graphics

**With patch 2, the following was also added:**

- Updated visuals for the Pogonia plant
- Improved pipe graphics
- Improved chain tiles
- Added outlines to the Halloween tileset
- Add spikes for the Ruin tileset
- Improved looks + new animation for unstable tiles
- Add missing outlines to Retro tilesets
- Minor changes to the pumpkin house
- Add autotiling for Halloween, Ice Chunk and Snow Fort tileset
- Fixed error hiding corrupted plant tile
- Removed several tiles from tilegroups which are going to be heavily changed or removed entirely soon

**With patch 3, the following was also added:**

- Update visuals for weakblocks (i.e. meltbox and strawbox)
- Improve and tweaked igloo decals
- Update poles
- New goalpost variant for secret level exits
- New flag design + animation
- Improve and new crystal background elements
- New bridge tilesets (frozen and un-frozen)
- Add missing retro snow tiles for autotiling (will follow soon)
- Add outlines to grave decals
- Move waterfall tiles over to 'images/tiles/water' + adjust colors
- Update retro brick and boxes to better suit art-style
- Update clouds
- Update rope, pushbuttons, snow bumps and forest walljump tiles - thanks to @weluvgoatz 
- Better magicblock visuals
- Add proper rublight graphics (crystal themed)
- Remove more outdated/un-wanted assets

**With patch 4, the following was also added:**

- Improved and expanded snow + ice cave tileset
- Updated retro tilesets
- Improved flying platforms
- Fixed crystal ground slopes
- Removed Snow Mountain tilegroup from tileset menu
- Removed even more old/unneeded graphics
- Updated lanterns + candles + torches
- Updated canon + dropper

---

**TODO List**

---

_Note: Theses changes will break some of the background assets in all levels using the older tilesets! Those will require manual updating!_

### CODE REVIEW

**Comment by Rusty-Box** [comment]
> were roots.png added back?

If you don't consider the commit "Re-add underground root tiles" a lie, then yes :D

**Comment by Rusty-Box** [comment]
A first "patch" for the ever expanding tileset changes is up now!

---

**This patch inlcudes the following:**

- Updated snow surface tileset
- Improved ice cave tileset
- Theme specific block overlays
- Updated castle tilesets
- New Ruins tileset
- Wall torch variant for torch object
- Floor tile for goal post
- Normal and corrupted flowers for the forest theme
- New sign to indicate the requirement of bombs
- Removal of more old/unneeded graphics

**Comment by Semphriss** [comment]
From #2175 : Reminder to do secret goal posts :stuck_out_tongue:

File: data/images/tiles.strf [hunk 1]
```diff
@@ -304,81 +306,65 @@
...
-      0    4268 4269 0
-      4254 4275 4276 4255
-      4262 4280 4281 4263
-      4270 4271 4272 4273
-      0    4277 4278 0
-      4282 4283 4284 4285
-      4288 4289 4290 4291
-      4286 0    0    0
-      4292 0    0    0
+      2647 2648 2649 2665
```
Comment (jonfx1007) [inline data/images/tiles.strf:350]:
```suggestion
      2655 2647 2656 2648

```

File: data/images/tiles.strf [hunk 2]
```diff
@@ -304,81 +306,65 @@
...
+	  2650 2651 2652 2666
```
Comment (jonfx1007) [inline data/images/tiles.strf:351]:
```suggestion
	  2659 2651 2660 2652

```

File: data/images/tiles.strf [hunk 3]
```diff
@@ -304,81 +306,65 @@
...
+	  2653 2654 2655 2663
```
Comment (jonfx1007) [inline data/images/tiles.strf:352]:
```suggestion
	  2662 2664 2661 2663

```

File: data/images/tiles.strf [hunk 4]
```diff
@@ -304,81 +306,65 @@
...
+	  2656 2657 2658 2664
```
Comment (jonfx1007) [inline data/images/tiles.strf:353]:
```suggestion
	  2658 2667 2657 2666

```

File: data/images/tiles.strf [hunk 5]
```diff
@@ -304,81 +306,65 @@
...
+	  2659 2660 0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:354]:
```suggestion
	  2649 2650 0    0

```

File: data/images/tiles.strf [hunk 6]
```diff
@@ -304,81 +306,65 @@
...
+	  2661 2662 0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:355]:
```suggestion
	  2653 2654 0    0

```

File: data/images/tiles.strf [hunk 7]
```diff
@@ -304,81 +306,65 @@
...
+	  
+	  0    2667 0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:357]:
```suggestion
	  0    2665 0    0

```

File: data/images/tiles.strf [hunk 8]
```diff
@@ -420,76 +412,44 @@
...
 	  
-      1358 1359 1360 0
-      1361 1362 1363 0
-      1364 1365 1366 0
-      458  459  460  0
-      461  462  463  0
-      464  465  466  0
-      467  468  469  0
-      470  471  472  0
+	  1746 1747 1748 0
```
Comment (jonfx1007) [inline data/images/tiles.strf:426]:
```suggestion
	  1760 1761 1749 0

```

File: data/images/tiles.strf [hunk 9]
```diff
@@ -420,76 +412,44 @@
...
+	  1749 1750 1751 0
```
Comment (jonfx1007) [inline data/images/tiles.strf:427]:
```suggestion
	  1754 1756 1757 0

```

File: data/images/tiles.strf [hunk 10]
```diff
@@ -630,32 +542,342 @@
...
+      0    1206 1207 0
+	  0    1213 1214 0
+	  1220 1221 1222 0
+	  1229 1230 1231 0
+	  1238 1239 1240 0
+	  1247 1248 1249 0
+	  1198 0    0    0
+	  1204 1205 0    0
+      1211 1212 0    0
+	  1218 1219 0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:606]:
```suggestion
	  1218 1219 2586    0

```

File: data/images/tiles.strf [hunk 11]
```diff
@@ -630,32 +542,342 @@
...
+	  1179 1180 1181 1182
+	  1184 0    0    0
+	  1186 0    0    0
+	  1188 0    0    0
+	  1190 0    0    0
+	  
+	  0    0    1194 0
+	  0    1199 1200 0
+      0    1206 1207 0
+	  0    1213 1214 0
```
Comment (jonfx1007) [inline data/images/tiles.strf:598]:
```suggestion
	  2583 1213 1214 0

```

File: data/images/tiles.strf [hunk 12]
```diff
@@ -518,85 +478,33 @@
...
-      1287 1288 1303 1304
-      1289 1290 1305 1306
-      1291 1292 1307 1308
-      1293 1294 1309 1310
-      1295 1296 0    0
-      1297 1298 0    0
-      1299 1300 0    0
-      1301 1302 0    0
-      402  403  404  405
+      4339 4340 2005 2006
```
Comment (jonfx1007) [inline data/images/tiles.strf:492]:
```suggestion
      4339 4340 1353 1354
      
```

File: data/images/tiles.strf [hunk 13]
```diff
@@ -518,85 +478,33 @@
...
+	  4337 4338 2007 2008
```
Comment (jonfx1007) [inline data/images/tiles.strf:493]:
```suggestion
	  2009 2014 4356 2008

```

File: data/images/tiles.strf [hunk 14]
```diff
@@ -518,85 +478,33 @@
...
+	  2009 2010 2011 2012
```
Comment (jonfx1007) [inline data/images/tiles.strf:494]:
```suggestion
	  4337 4334 4348 4344

```

File: data/images/tiles.strf [hunk 15]
```diff
@@ -518,85 +478,33 @@
...
+	   473  474  475 2013
```
Comment (jonfx1007) [inline data/images/tiles.strf:495]:
```suggestion
	   473 1821  474 2013

```

File: data/images/tiles.strf [hunk 16]
```diff
@@ -518,85 +478,33 @@
...
+	   476  0   4356 4353
```
Comment (jonfx1007) [inline data/images/tiles.strf:496]:
```suggestion
	  2012  0   2007 4353

```

File: data/images/tiles.strf [hunk 17]
```diff
@@ -518,85 +478,33 @@
...
+	   477  478 4357 4354
```
Comment (jonfx1007) [inline data/images/tiles.strf:497]:
```suggestion
	  4338 4333 4357 4354

```

File: data/images/tiles.strf [hunk 18]
```diff
@@ -518,85 +478,33 @@
...
+	  4333 4334 4341 4342
```
Comment (jonfx1007) [inline data/images/tiles.strf:498]:
```suggestion
	  2005 2010 4341 4342

```

File: data/images/tiles.strf [hunk 19]
```diff
@@ -518,85 +478,33 @@
...
+	  4335 4336 4349 4350
+	  4358 4359 4332 4345
+	  4348 0    4347 4346
```
Comment (jonfx1007) [inline data/images/tiles.strf:501]:
```suggestion
	  2011 0    2006 4346

```

File: data/images/tiles.strf [hunk 20]
```diff
@@ -518,85 +478,33 @@
...
+	  4343 101  4344 4355
```
Comment (jonfx1007) [inline data/images/tiles.strf:502]:
```suggestion
	   475 1351  476 4355

```

File: data/images/tiles.strf [hunk 21]
```diff
@@ -518,85 +478,33 @@
...
+	  1352 2014 1353 1351
```
Comment (jonfx1007) [inline data/images/tiles.strf:503]:
```suggestion
	   477 1352  478  101

```

File: data/images/tiles.strf [hunk 22]
```diff
@@ -518,85 +478,33 @@
...
+	  1354 1821 4351 4352
```
Comment (jonfx1007) [inline data/images/tiles.strf:504]:
```suggestion
	  4347 4343 4351 4352

```

File: data/images/tiles.strf [hunk 23]
```diff
@@ -630,32 +542,342 @@
...
+	  0    0    1121 1122
+	  0    0    1123 1124
+	  1279 0    0    0
+	  1282 1283 0    0
+	  1286 1287 0    0
+	  1290 1291 0    0
+	  1294 1295 0    0
+	  1299 1300 1301 0
+	  1305 1306 1307 0
+	  1310 1311 0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:820]:
```suggestion
	  1310 1312 0    0

```

File: data/images/tiles.strf [hunk 24]
```diff
@@ -671,7 +893,7 @@
 	  102  140  3161 3162
 	  104  105  3160  112
 	  3037 2943 2944 2945
-	  2946 1311 2153 4434
+	  2946 1312 4434 0
```
Comment (jonfx1007) [inline data/images/tiles.strf:896]:
```suggestion
	  2946 1311 4434 0

```

File: data/images/tiles.strf [hunk 25]
```diff
@@ -762,167 +1010,116 @@
...
-	    3280 3277 3278 0
-	    3257 3258 3269 3270
-      3259 3260 3271 3272
+      2106 2092 2107 2108
+	  2096 2112 2097 2098
+	  2116 2102 2117 2118
+	  2111 2113 2095 2094
+	  1752 2104 2101 2103
+	  1753 2100 3515 2105
+	  1754 1755 2091 2093
```
Comment (jonfx1007) [inline data/images/tiles.strf:1035]:
```suggestion
	  1748 1755 2091 2093

```

File: data/images/tiles.strf [hunk 26]
```diff
@@ -762,167 +1010,116 @@
...
-      3890 3891 3892 3893
-      3894 3895 3896 3897
-      3898 3899 3900 3901
-      3902 3903 3904 3905
-      3906 3907 3908 3909
-      3910 0    3911 0
-      3912 3913 3914 0
-      3915 3916 3917 3918
-      3919 3920 3921 3922
+	  3888 3889 1756 1758
```
Comment (jonfx1007) [inline data/images/tiles.strf:1038]:
```suggestion
	  3888 3889 1750 1758

```

File: data/images/tiles.strf [hunk 27]
```diff
@@ -762,167 +1010,116 @@
...
+	  3890 3891 1757 1759
```
Comment (jonfx1007) [inline data/images/tiles.strf:1039]:
```suggestion
	  3890 3891 1751 1759

```

File: data/images/tiles.strf [hunk 28]
```diff
@@ -762,167 +1010,116 @@
...
+	  3892 3893 1760 1761
```
Comment (jonfx1007) [inline data/images/tiles.strf:1040]:
```suggestion
	  3892 3893 1746 1747

```

File: data/images/tiles.strf [hunk 29]
```diff
@@ -762,167 +1010,116 @@
...
-	    2091 2092 2093    0
-      2096 2097 2098    0
-      2101 2102 2103    0
-      2106 2107 2108    0
-      2111 2112 2113    0
-      2116 2117 2118 2099
-      2109 2104 2110 2100
-      2094 2095 2105 3515
-      2114 2115 2119 2120
+	  1779 1780 1781 1783
```
Comment (jonfx1007) [inline data/images/tiles.strf:1046]:
```suggestion
	  1779 1780 1781 1802

```

File: data/images/tiles.strf [hunk 30]
```diff
@@ -762,167 +1010,116 @@
...
+	  1784 1785 1786 1798
```
Comment (jonfx1007) [inline data/images/tiles.strf:1047]:
```suggestion
	  1784 1787 1795 1798

```

File: data/images/tiles.strf [hunk 31]
```diff
@@ -762,167 +1010,116 @@
...
+	  1789 1790 1792 1793
```
Comment (jonfx1007) [inline data/images/tiles.strf:1048]:
```suggestion
	  1801 1790 1792 1783

```

File: data/images/tiles.strf [hunk 32]
```diff
@@ -762,167 +1010,116 @@
...
+	  1794 1788 1807 1806
+	  1801 1782 1799 1800
```
Comment (jonfx1007) [inline data/images/tiles.strf:1050]:
```suggestion
	  1775 1782 1799 1800

```

File: data/images/tiles.strf [hunk 33]
```diff
@@ -762,167 +1010,116 @@
...
+	  1802 1795 1805 1804
```
Comment (jonfx1007) [inline data/images/tiles.strf:1051]:
```suggestion
	  1774 1797 1805 1804

```

File: data/images/tiles.strf [hunk 34]
```diff
@@ -762,167 +1010,116 @@
...
+	  1774 1776 1796 1797
```
Comment (jonfx1007) [inline data/images/tiles.strf:1052]:
```suggestion
	  1793 1776 1803 1808

```

File: data/images/tiles.strf [hunk 35]
```diff
@@ -762,167 +1010,116 @@
...
+	  1775 0    0    0
```
Comment (jonfx1007) [inline data/images/tiles.strf:1053]:
```suggestion
	  1789 0    0    0

```

File: data/images/tiles.strf [hunk 36]
```diff
@@ -762,167 +1010,116 @@
...
-	    1779 1782 1802    0
-      1784 1795 1798    0
-      1799 1797 1800    0
-      1801 1781 1783 1780
-      1794 1787 1788    0
-      1786 1791 1796    0
-      1790 1792 1789 1793
-      1804 1805 1806 1807
+	  2265 3912 3913 3914
+	  1777 1778 1803 1808
```
Comment (jonfx1007) [inline data/images/tiles.strf:1056]:
```suggestion
	  1791 1785 1796 1786

```

File: data/images/tiles.strf [hunk 37]
```diff
@@ -762,167 +1010,116 @@
...
-      0    1397 1398 1399
-      1400 1401 1402 1403
-      1404 1405 1406 1407
-      1408 1409 1410 0
-      1412 1413 1414 1415
-      1416 1417 1419 0
-      0    1418 1420 0
-      1467 1468 1469 1470
+	  1762 1763 3906 3907
+	  1764 1765 3908 3909
```
Comment (jonfx1007) [inline data/images/tiles.strf:1059]:
```suggestion
	  3269 3270 3908 3909

```

File: data/images/tiles.strf [hunk 38]
```diff
@@ -762,167 +1010,116 @@
...
+	  1766 1767 1787 1791
```
Comment (jonfx1007) [inline data/images/tiles.strf:1060]:
```suggestion
	  3259 3260 1778 1777

```

File: data/images/tiles.strf [hunk 39]
```diff
@@ -762,167 +1010,116 @@
...
+	  1768 1769 3903 3904
```
Comment (jonfx1007) [inline data/images/tiles.strf:1061]:
```suggestion
	  3271 3272 3903 3904

```

File: data/images/tiles.strf [hunk 40]
```diff
@@ -762,167 +1010,116 @@
...
+	  1770 1771 3910 3911
```
Comment (jonfx1007) [inline data/images/tiles.strf:1062]:
```suggestion
	  1772 1773 3910 3911

```

File: data/images/tiles.strf [hunk 41]
```diff
@@ -762,167 +1010,116 @@
...
+	  3898 3899 3900 3901
+	  3902 0    0    3905
+	  1772 1773 2214 2215
```
Comment (jonfx1007) [inline data/images/tiles.strf:1065]:
```suggestion
	  2212 2213 2214 2215

```

File: data/images/tiles.strf [hunk 42]
```diff
@@ -762,167 +1010,116 @@
...
+	  3257 3258 2216 2217
+	  3269 3270 2218 2219
```
Comment (jonfx1007) [inline data/images/tiles.strf:1067]:
```suggestion
	  1764 1765 2218 2219

```

File: data/images/tiles.strf [hunk 43]
```diff
@@ -762,167 +1010,116 @@
...
+	  3259 3260 2220 2221
```
Comment (jonfx1007) [inline data/images/tiles.strf:1068]:
```suggestion
	  1766 1767 2220 2221

```

File: data/images/tiles.strf [hunk 44]
```diff
@@ -762,167 +1010,116 @@
...
+	  3271 3272 2222 2223
```
Comment (jonfx1007) [inline data/images/tiles.strf:1069]:
```suggestion
	  1768 1769 2222 2223

```

File: data/images/tiles.strf [hunk 45]
```diff
@@ -762,167 +1010,116 @@
...
+	  2212 2213 2224 2225
```
Comment (jonfx1007) [inline data/images/tiles.strf:1070]:
```suggestion
	  1770 1771 2224 2225

```

File: data/images/tiles.strf [hunk 46]
```diff
@@ -762,167 +1010,116 @@
...
+	  2259 2260 0    0
+	  
+	  2226 2227 2232 2234
+	  2228 2229 2233 2235
+	  2236 2237 2241 2243
+	  2238 2239 2242 2244
+	  2245 2246 2250 2252
+	  2247 2248 2251 2253
+	  2230 2231 2240 2249
+	  295  296  297  298
```
Comment (jonfx1007) [inline data/images/tiles.strf:1080]:
```suggestion
	  297  295  298  296

```

File: data/images/tiles.strf [hunk 47]
```diff
@@ -762,167 +1010,116 @@
...
+	  
+	  2257 1489 2255 0
+	  1482 1476 2254 0
+	  2258 1496 2256 0
+	  1478 1477 2261 2262
```
Comment (jonfx1007) [inline data/images/tiles.strf:1085]:
```suggestion
	  2263 2264 2261 2262

```

File: data/images/tiles.strf [hunk 48]
```diff
@@ -762,167 +1010,116 @@
...
+	  1503 1510 2263 2264
```
Comment (jonfx1007) [inline data/images/tiles.strf:1086]:
```suggestion
	  1503 1510 1477 1478

```

**Review by jonfx1007 — CHANGES_REQUESTED** [review]
After playing around with the tileset update patch for a good while, I have a number of suggestions to increase retrocompatibility by trying to preserve old tile structures as much as possible. You can see the changes to tile IDs line-by-line in the tiles.strf file. Sorry if it gets a bit annoying; I didn't see a good way to do multi-line suggestions in Github. 
Obviously it isn't possible to preserve all old tiles; for example, the forest trees can't be salvaged. But I tried to avoid breaking tiles where possible.
The autotiles.satc file will still need to be updated; I'm willing to help with that if/when changes to tile IDs are accepted.

**Pull request merged**